lib/dal/src/cached_module.rs

```diff
@@ -230,12 +277,24 @@ impl CachedModule {
         let ctx_clone = ctx.clone();
         ctx_clone.commit_no_rebase().await?;
 
-        let hashes: Vec<_> = modules.keys().map(ToOwned::to_owned).collect();
+        let new_modules = Self::cache_modules(ctx, &modules, module_index_client).await?;
```

Comment:
Moved cache_modules to its own method for clarity instead of dumping my new stuff into the method along with it.